GoogleForCreators/web-stories-wp
# Redesign: Animation Dropdown

## Context

Building off #6505 which redesigns the animation panel to refactor the custom effect dropdown to use shared materials in design system. 

## Summary

Guts the custom dropdown we were using for animation effects. Should look like this: https://www.figma.com/file/bMhG3KyrJF8vIAODgmbeqT/Design-System?node-id=221%3A13978 


## Relevant Technical Choices

- Moved files that create effect dropdown into their own subdirectory - `/animation` was getting too big. 
- Restructured effect data into background and foreground objects that we can use to map through to create our dropdown options and still have access to all the custom values that each animation needs to preview and be selected. 
- Renamed some props internal to the dropDown's `Menu` to quiet some eslint warnings. Fix prop for `dropDownHeight` so that it can be overwritten. 


## To-do

- There's a side convo happening about how to handle the scrollbar in this dropdown since the designs don't account for it. 

## User-facing changes

Should look the same as in #6505
except i shrank some of the fonts a bit to avoid crowding and flipped the foreground dropdown to open up to give it more room.

https://user-images.githubusercontent.com/10720454/112236399-994ed800-8bfd-11eb-9295-c07e9b5cf2ba.mp4


https://user-images.githubusercontent.com/10720454/112236400-99e76e80-8bfd-11eb-95d7-cdfd5b6532f6.mp4



## Testing Instructions

Verify that selecting animation effects has no regression: 
- Select a foreground element, apply different animations to it with the dropDown. See that dropDown works with both cursor and keyboard. 
- Select background image, apply different animations to it with the dropDown. See that effects are disabled with a tooltip telling you why in some cases. 

### QA






This PR can be tested by following these steps:

1.

### UAT







This PR can be tested by following these steps:

1.

## Reviews

### Does this PR have a security-related impact?

No

### Does this PR change what data or activity we track or use?

No

### Does this PR have a legal-related impact?

No

## Checklist




---



Fixes #6827
